custom/icds_reports/static/js/directives/info-message/info-message.directive.js

```diff
@@ -6,30 +6,57 @@ function InfoMessageController($location) {
     var vm = this;
     vm.previousToPreviousMonth = null;
     vm.previousMonth = null;
+    vm.yearOfPreviousMonth = null;
+    vm.yearOfTwoMonthsAgo = null;
     vm.currentMonth = null;
     vm.nextMonth = null;
+    vm.yearOfNextMonth = null;
     vm.year = null;
 
+    vm.getMonthFromDate = function (date) {
+        return moment().month(date.getMonth()).format('MMMM');
+    };
+
+    vm.getYearFromDate = function (date) {
+        return moment().month(date.getMonth()).year(date.getFullYear()).format('YYYY');
+    };
+
+
     vm.showInfoMessage = function () {
         var start = parseInt(vm.start || '1');
         var end = parseInt(vm.end || '2');
         vm.type = vm.type || 'default';
 
         var selectedMonth = parseInt($location.search()['month']) || new Date().getMonth() + 1;
         var selectedYear = parseInt($location.search()['year']) || new Date().getFullYear();
-        var currentMonth = new Date().getMonth() + 1;
-        var currentYear = new Date().getFullYear();
-        var currentDate = new Date().getDate();
+        var now = new Date();
+        var currentMonth = now.getMonth() + 1;
+        var currentYear = now.getFullYear();
+        var currentDate = now.getDate();
 
         var displayInfoMessage = selectedMonth === currentMonth && selectedYear === currentYear && (currentDate >= start && currentDate <= end);
 
         if (displayInfoMessage && !$location.path().startsWith("/download")) {
-            vm.previousMonth = moment().startOf('month').subtract(1, 'months').format('MMMM');
-            vm.previousToPreviousMonth = moment().startOf('month').subtract(2, 'months').format('MMMM');
             vm.currentMonth = moment().format("MMMM");
-            vm.nextMonth = moment().startOf('month').add(1, 'months').format("MMMM");
             vm.year = moment().format('YYYY');
 
+            var previousMonthDate = new Date();
+            var previousToPreviousMonthDate = new Date();
+            var nextMonthDate = new Date();
+
+            previousMonthDate.setMonth(now.getMonth() - 1);
+            previousToPreviousMonthDate.setMonth(now.getMonth() - 2);
```

Comment:
TIL: this allows negative values and works. https://www.w3schools.com/jsref/jsref_setmonth.asp